packages/toolpad-core/src/DashboardLayout/DashboardLayout.tsx

```diff
@@ -149,6 +152,12 @@ function DashboardLayout(props: DashboardLayoutProps) {
     sx,
   } = props;
 
+  if (navigationProp && process.env.NODE_ENV === 'development') {
```

Comment:
We almost exclusively check against `'production'` in our codebases

```suggestion
  if (navigationProp && process.env.NODE_ENV !== 'production') {
```